pubsub_nats.go

```diff
@@ -165,9 +165,8 @@ func (p *natsPubSub) Subscribe(pubs chan *Publication, errors chan error, topic
 			// to whenever it is ready. The subscriber SHOULD attempt to respond ASAP as there is a client waiting
 			// for a response.
 			case ResponseModePublication:
-				replyCh := make(chan *Publication)
-				publication.replyCh = replyCh
-				go responseHandler(m.Reply, replyCh)
+				publication.replyCh = make(chan *Publication)
```

Comment:
you are not using the inner lock here. add a method `setReplyCh()` to publication that locks/unlocks

Reply:
there is no race condition at this point because this is the only goroutine active that has access to this variable.

You can run the test suite with `-race` to verify.